OutSystems/outsystems-datagrid
# ROU-4344: remove dirty marks from action column

This PR is for fixing a misbehavior that caused the action column to get dirty marks in some situations.
![datagrid-action-dirty](https://github.com/OutSystems/outsystems-datagrid/assets/10534623/482b1fe7-f499-48dd-b7f4-2fb652725fe7)

### What was happening
* When an action column was bound to another column, and that column was changed, since they will have the same binding the code considered that both had the dirty mark.
* Action columns, cannot however be changed, so the dirty mark would be displayed, even thought it would not make sense.

### What was done
* Added a new method to the `IGrid`, named `getColumnByIndex`, that receives an index, and returns the column on that index
* The dirty mark will now consider the type of the column
* If the cell belongs to a column of the type action, then it will always be considered as not dirty

### Test Steps
1. Go to a page with an editable grid, with multiple columns, with some action columns with shared binding
2. Change the value of a cell in one of the editable columns
3. The dirty mark should appear in the edited cell, but not in the associated action column


### Screenshots after fix
![datagrid-action-dirty-fixed](https://github.com/OutSystems/outsystems-datagrid/assets/10534623/0daafeed-37f2-49e2-8461-6a7e0e67e37b)

### Checklist
